test/sink.test.js

```diff
@@ -42,3 +45,113 @@ test('.writer() - no value for "type" argument - should throw', () => {
         sink.writer();
     }).toThrowError('"type" is missing');
 });
+
+test.only('.writer() - happy path', async done => {
+    expect.assertions(2);
+    const sink = getValidSink();
+
+    await sink.ready;
+
+    const gcsFakeWritestream = new stream.Writable({
+        _data: false,
+        write(chunk, encoding, next) {
+            this._data += chunk;
+            next();
+        },
+    });
+
+    sink.gcs._setState({
+        createWriteStream: gcsFakeWritestream,
+        move: null,
+    });
+
+    const dest = sink.writer('json');
+
+    const source = require('fs').createReadStream(
+        `${__dirname}/mock/feed.a.json`
+    );
+
+    source.on('error', done);
+    dest.on('error', done);
+    gcsFakeWritestream.on('error', done);
+
+    source.pipe(dest);
+
+    dest.on('file not saved', () => {
+        done(new Error('File not saved'));
+    });
+
+    dest.on('file saved', (id, file) => {
+        expect(id).toMatchSnapshot();
+        expect(file).toMatchSnapshot();
+        process.nextTick(done);
```

Comment:
I think i've seen this somewhere else as well, wondering why do we need to do this?